packages/jest-types/src/Config.ts

```diff
@@ -254,6 +254,72 @@ type ShardConfig = {
   shardCount: number;
 };
 
+export type PluginEnforce = 'pre' | 'post';
+
+export type PluginContext = {
+  config: ProjectConfig;
+  globalConfig: GlobalConfig;
+};
+
+export type ResolveIdResult =
+  | {
+      id: string;
+      external?: boolean;
+    }
+  | string
+  | null
+  | undefined;
+
+export type LoadResult =
+  | {
+      code: string;
+      map?: any;
+    }
+  | string
+  | null
+  | undefined;
+
+export type TransformPluginResult =
+  | {
+      code: string;
+      map?: any;
+    }
+  | string
+  | null
+  | undefined;
+
+export interface JestPlugin {
+  name: string;
+  enforce?: PluginEnforce;
+  config?: (
+    config: InitialOptions,
+    context: {configPath: string | null},
+  ) => InitialOptions | Promise<InitialOptions> | void | Promise<void>;
+  configResolved?: (
+    config: ProjectConfig,
+    context: PluginContext,
+  ) => void | Promise<void>;
+  configureJest?: (context: PluginContext) => void | Promise<void>;
+  resolveId?: (
+    source: string,
+    importer: string | undefined,
+    options: {isEntry: boolean},
+  ) => ResolveIdResult | Promise<ResolveIdResult>;
+  load?: (id: string) => LoadResult | Promise<LoadResult>;
+  transform?: (
```

Comment:
I would like `jest-transform` to use this `transform` property when plugins are defined. If no plugins are defined, `jest-transform` can reuse its existing logic to transform codes.

Currently, Jest transforms codes only on project level while Vitest has its at root. I would like to also add the possibility to transform codes on global level if no `projects` is specified.